### PR TITLE
Alt text on ImageGrid block

### LINKED ISSUE #4401: Image grid content block doesn't allow alt-text
When using the 'Image Grid' content block in wagtail, I'm unable to add alt-text to the images in the grid. 

It currently uses the file name, which seems to also be autogenerated with the new image size. 

Screenshot 2020-03-24 at 15 44 36

Would it be possible to add in the alt-text option for this content block?

(This was on the MozFest site, however I imagine it would be the same on Foundation site but I haven't tested.)

### PR DESCRIPTION
**Pull request by KalobTaulien**
# ❗ Do not merge until `ia-refresh-homepage` is merged in

Closes #4401 

**Note**: Prior to merging into `master` let's make sure the migrations in this branch are re-created on top of the `ia-refresh-homepage` migrations.

### CODE REVIEW

File: network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/figure_block2.html [hunk 1]
```diff
@@ -5,9 +5,9 @@
 <div class="col-6 {% if count > 3 %}col-sm-3{% elif count == 3 %}col-sm-4{% elif count == 2 %}col-sm-6{% endif %} mb-2 mb-sm-4">
 	{% if url %}<a href="{{ url }}">{% endif %}
     {% if square %}
-      {% image value.image fill-890x890-c100 format-jpeg class="w-100" %}
+      {% image value.image fill-890x890-c100 format-jpeg class="w-100" alt=value.alt_text %}
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/figure_block2.html:8]: we'll need some fallbacks here

Comment (KalobTaulien) [reply network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/figure_block2.html:8]: Fair point. Will work on this shortly. Thanks!

**Comment by KalobTaulien** [comment]
@Pomax Would you be opposed to having figure_block2.html renamed to figure_block.html?

File: network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/figure_block2.html (hunk 1 above)
Comment (KalobTaulien) [reply network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/figure_block2.html]: Added a StructValue to fallback the alt text to the image title in https://github.com/mozilla/foundation.mozilla.org/pull/5041/commits/652c1084b5c933e1fa5d3892b8402ca0f613736b

File: network-api/networkapi/wagtailpages/pagemodels/customblocks/image_grid.py [hunk 2]
```diff
@@ -18,6 +33,9 @@ class ImageGrid(blocks.StructBlock):
         help_text='If left checked, the image will be cropped to be square.'
     )
 
+    class Meta:
+        value_class = ImageGridAltText
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/pagemodels/customblocks/image_grid.py:37]: I keep forgetting this is a thing. It's kind of deep wagtail/django magic, but a nice solution here.

File: network-api/networkapi/wagtailpages/pagemodels/customblocks/image_grid.py [hunk 3]
```diff
@@ -2,8 +2,23 @@
...
+        image = self.get('image')
+        if alt_text:
+            return alt_text
+        return image.title
+
+
 class ImageGrid(blocks.StructBlock):
     image = ImageChooserBlock()
+    alt_text = blocks.CharBlock(
+        required=False,
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/pagemodels/customblocks/image_grid.py:19]: This looks like it's not working. Do we need `blank=True` instead?

Comment (Pomax) [reply network-api/networkapi/wagtailpages/pagemodels/customblocks/image_grid.py:19]: hm, tried `blank=True` and `null=True`, but the CMS still goes "nope, I _need_ you to fill in an alt text"

Comment (Pomax) [inline network-api/networkapi/wagtailpages/pagemodels/customblocks/image_grid.py:19]: tested this on a campaign page by adding an image grid, but publishing still says I have to fill something in:
image

File: network-api/networkapi/wagtailpages/pagemodels/customblocks/image_grid.py (hunk 2 above)
Comment (KalobTaulien) [reply network-api/networkapi/wagtailpages/pagemodels/customblocks/image_grid.py:37]: Thanks!

**Pull request merged**